integration_tests/src/main/python/data_gen.py

```diff
@@ -351,9 +372,12 @@ def _loop_values(self):
 
     def start(self, rand):
         self._index = 0
-        self._child.start(rand)
         self._start(rand, self._loop_values)
-        self._vals = [self._child.gen() for _ in range(0, self._length)]
+        if isinstance(self._child, list):
+            self._vals = self._child
```

Comment:
length is not honored here:
```suggestion
            self._vals = self._child[:length]
```